### PR TITLE
fix/error-while-confirm-booking

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Adds try/catch when confirming a booking that has workflow.
- Adds toast for success/error while confirming

Fixes # (issue)

**Environment**: Staging(main branch)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Make workflow fail and now you should see confirmation works but an error on console.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -44,6 +44,11 @@ function BookingListItem(booking: BookingItemProps) {
   const mutation = trpc.useMutation(["viewer.bookings.confirm"], {
     onSuccess: () => {
       setRejectionDialogIsOpen(false);
+      showToast(t("booking_confirmation_success"), "success");
+      utils.invalidateQueries("viewer.bookings");
+    },
+    onError: () => {
+      showToast(t("booking_confirmation_failed"), "error");
```
Comment (alannnc) [inline apps/web/components/booking/BookingListItem.tsx:51]: Now notify the user when succeeded/failed.

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 2]
```diff
@@ -461,37 +461,48 @@ export const bookingsRouter = createProtectedRouter()
...
+          subscribersMeetingEnded.forEach((subscriber) => {
+            updatedBookings.forEach((booking) => {
+              scheduleTrigger(booking, subscriber.subscriberUrl, subscriber);
+            });
           });
-        });
+        } catch (error) {
+          // Silently fail
+          console.error(error);
+        }
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/bookings.tsx:505]: Workflows and reminders can fail but we should still let the user know if booking confirmation was successful or now.

**Review by zomars — APPROVED** [review]
Good enough for now, we should discuss the optimal way to handle Workflow errors and maybe prompt the user for action to resolve it. cc @CarinaWolli

**Comment by zomars** [comment]
Pushed a revert that fixes E2E tests. Waiting for checks to pass.

**Comment by zomars** [comment]
Will merge as there's a flaky test unrelated to this PR

**Pull request merged**